Onboarding updated

johnwatson484 opened the pull request:
https://eaflood.atlassian.net/browse/PSD-700

johnwatson484 committed:
Onboarding updated

johnwatson484 committed:
Merge remote-tracking branch 'origin/master' into psd-700-onboarding

johnwatson484 committed:
CSC renamed

johnwatson484 committed:
Pipeline added

johnwatson484 committed:
Merge remote-tracking branch 'origin/master' into psd-700-onboarding

johnwatson484 committed:
Pipeline updated

johnwatson484 committed:
Webhook updated

johnwatson484 committed:
Link to library added

st3v3nhunt commented on guides/resource-provisioning/ci-pipeline.md:
```diff
@@ -0,0 +1,44 @@
+- enter `GitHub` as a Branch Source
+- for credentials select the `ffcplatform` GitHub token
+- enter your GitHub URL in the `HTTPS URL` field, for example `https://github.com/DEFRA/ffc-demo-web.git`
+- set `Pipeline Action Triggers -> Pipeline Delete Event` set `ffc-housekeeping/cleanup-on-branch-delete` 
```
It might be worth adding a step to describe setting `Scan Repository Triggers -> Periodically if not otherwise run` to be `checked` and with an `Interval` of `1 hour`

st3v3nhunt commented on guides/resource-provisioning/ci-pipeline.md:
```diff
@@ -0,0 +1,44 @@
+
+### Configure GitHub repository
+
+- generate an SSH key pair
```
Is this on the dev's machine?

st3v3nhunt reviewed (commented):
The instructions are good, I've added a couple of comments about potential improvements/clarifications.
Do you want these instructions to be enough so that a first build will completely succeed? I ask as I've got the build failing on the `SonarCloud` step atm.

Is there going to be a section for adding the deploy pipeline too? I think it would be helpful.

johnwatson484 committed:
SonarCloud setup added

johnwatson484 committed:
Snyk added

johnwatson484 committed:
CI updated

johnwatson484 commented:
> The instructions are good, I've added a couple of comments about potential improvements/clarifications.
> Do you want these instructions to be enough so that a first build will completely succeed? I ask as I've got the build failing on the `SonarCloud` step atm.
> 
> Is there going to be a section for adding the deploy pipeline too? I think it would be helpful.

Thanks I've updated based on your feedback.

For release, yes there will be in a separate document which I'll add soon.  What we have for release at the moment is temporary and will be gone once we have our "proper" release pipelines in ADO.  But you're right, we should have something now otherwise the main branch build will at the `Trigger Deploy` step.

johnwatson484 committed:
Release pipeline added

johnwatson484 committed:
Release pipeline added

johnwatson484 committed:
Formatting updated

johnwatson484 committed:
formating updated

st3v3nhunt commented on guides/resource-provisioning/release-pipeline.md:
```diff
@@ -0,0 +1,41 @@
+- navigate to your workstream folder in Jenkins
+- select `Credentials -> (global) -> Add Credentials`
+- select `Secret text` type and set both `Description` and `Id` to `<respository name>-deploy-token`, for example `ffc-demo-web-deploy-token`
+- set the secret value to be a unique string
```
You mention this is only temporary so it might not be worth changing anything based on this suggestion. Either way, the suggestion is thus...
Having some additional context added to this bullet about what the string is and how it is used i.e. as a 'secret' token used for 'authenticating' (unnecessary quotes!) the triggering of the deploy job by the build job and the string is entered later in the deploy creation.

johnwatson484 committed:
Context added to token

johnwatson484 committed:
Fix grammar

FayToward reviewed (approved):
LGTM
